pkg/common/util/validation.go

```diff
@@ -0,0 +1,47 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+// ValidateSpiffeID verify than Spiffe Id  is valid  according to SVID spec,
+// validations done:
+// - spiffe id is not empty
+// - spiffe id is a valid url
+// - scheme is 'spiffe'
+// - host is not empty
+// - host belong to provided trust domain
+// - path is not empty
+// - path does not start with '/spire/' since it is reserved for agent, server, etc.
+func ValidateSpiffeID(spiffeID string, trustDomain url.URL) error {
+
+	// Validate Spiffe Id is provided
+	if spiffeID == "" {
+		return errors.New("a SPIFFE ID is required")
+	}
+
+	// Parse Spiffe Id to url
+	id, err := url.Parse(spiffeID)
+	if err != nil {
+		return fmt.Errorf("could not parse SPIFFE ID: %s", err.Error())
+	}
+
+	// Validate than spiffe id is valid
+	if id.Scheme != "spiffe" || id.Host == "" || id.Path == "" || strings.HasPrefix(id.Path, "/spire/") {
+		return fmt.Errorf("\"%s\" is not a valid SPIFFE ID", id.String())
+	}
+
+	if trustDomain.Host == "" {
+		return errors.New("a trust domain is required")
+	}
+
+	// Verify than provided spiffe id belong to trust domain
```

Comment:
Maybe: Verify _that_ the provided spiffe id _belongs_ to the configured trust domain

Reply:
done